back/zomes/projects/src/project/project_meta/validate.rs

```diff
@@ -0,0 +1,315 @@
+use crate::project::{
+  error::Error,
+  project_meta::crud::{inner_fetch_project_metas, ProjectMeta},
+  validate::validate_value_matches_create_author,
+};
+use dna_help::{resolve_dependency, ResolvedDependency};
+use hdk::prelude::*;
+
+#[hdk_extern]
+fn validate_create_entry_project_meta(
+  validate_data: ValidateData,
+) -> ExternResult<ValidateCallbackResult> {
+  Ok(
+    // element must have an entry that must deserialize correctly
+    match ProjectMeta::try_from(&validate_data.element) {
+      Ok(proposed_entry) => {
+        // `address` must match header author
+        match validate_value_matches_create_author(
+          &proposed_entry.creator_address.0,
+          &validate_data,
+        ) {
+          ValidateCallbackResult::Valid => {
+            // no project_meta entry should exist at least
+            // that we can know about
+            match inner_fetch_project_metas(GetOptions::content())?.0.len() {
```

Comment:
I think this is not really deterministic. I can see two failure scenarios:

- Alice and Bob create the project meta "at the same time".
  - In this case, local validation will pass because Alice only sees her own project meta, and Bob his own. 
  - Two Create headers with the same entry and two links will be published in the DHT
  - When they are published in the DHT, some peers could start with Alice's, some others with Bob's
  - When the first one is validated by the peers and they move on to validate the second one (for some validators that's Bob's, for some that's Alice's), they are going to mark it invalid -> non-determinism!

- After initial "fast push" validation, some validator gets the project meta Create element and has to validate it.
  - It does the get links and get inside the inner_fetch_project_metas
  - Since the other validators do have both the entry and link validated, they are going to return it
  - Validation is going to return invalid -> non-determinism!

This is what I think should happen at least :) . It could be fun to test it in a local network or even write unit/tryorama tests for these cases.

Reply:
as we spoke about, you're pretty much right, and what I was really wanting was the "runtime check" during a zome call, to invalidate the craetor of one from creating one, but only superficially. Good chat today 🙏 

Going to be updating this code